include/ebpf_nethooks.h

```diff
@@ -126,6 +130,56 @@ EBPF_HELPER(int, bpf_sock_addr_set_redirect_context, (bpf_sock_addr_t * ctx, voi
     ((bpf_sock_addr_set_redirect_context_t)BPF_FUNC_sock_addr_set_redirect_context)
 #endif
 
+/**
+ * @brief Get the interface type for the connection (AUTH_CONNECT and AUTH_RECV_ACCEPT only).
+ *
+ * @param[in] ctx Pointer to bpf_sock_addr_t context.
+ *
+ * @retval Interface type value, or 0 if not available.
+ */
+EBPF_HELPER(uint32_t, bpf_sock_addr_get_interface_type, (bpf_sock_addr_t * ctx));
+#ifndef __doxygen
+#define bpf_sock_addr_get_interface_type ((bpf_sock_addr_get_interface_type_t)BPF_FUNC_sock_addr_get_interface_type)
+#endif
+
+/**
+ * @brief Get the tunnel type for the connection (AUTH_CONNECT and AUTH_RECV_ACCEPT only).
+ *
+ * @param[in] ctx Pointer to bpf_sock_addr_t context.
+ *
+ * @retval Tunnel type value, or 0 if not a tunnel or not available.
```

Comment:
How will the caller differentiate between "not a tunnel" and "not available".

One option can be to change the return type to int32 and return -1 when the helper function is called for an incorrect attach type. That helps differentiate the above 2 cases.

Same for all the new helper functions.